### ISSUE TITLE
Dependency issues with pendulum and python 3.12

### ISSUE THREAD

**Issue by cldmkr** [dusktreader/flask-praetorian#292]
flask-praetorian requires package pendulum `^2.1`.
Pendulum <3.0.0 requires package distutils. Package distutils is no longer available on python 3.12.
Package pendulum removed the dependency to distutils starting with version 3.0.0 (Ref: [Pendulum issues 696](https://github.com/sdispater/pendulum/issues/696)).

To be able to use flask-praetorian with python 3.12 and later, I assume flask-praetorian has to support pendulum >= 3.0.0.

```
>> pip install --upgrade flask-praetorian==1.5.0 pendulum==3.0.0
[...]
Collecting pendulum==3.0.0
  Using cached pendulum-3.0.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.9 kB)
[...]
INFO: pip is looking at multiple versions of flask-praetorian to determine which version is compatible with other requirements. This could take a while.
Collecting flask-praetorian==1.5.0
  Using cached flask_praetorian-1.5.0-py3-none-any.whl.metadata (4.0 kB)
ERROR: Cannot install flask-praetorian==1.5.0 and pendulum==3.0.0 because these package versions have conflicting dependencies.

```

**Comment by dusktreader** [comment]
Addressed by PR #296

**Comment by dusktreader** [comment]
@cldmkr 
Support for Python 3.12 has been published in flask-praetorian v1.6.0 on PyPI.
